x-pack/plugins/observability_solution/inventory/public/components/badge_filter_with_popover/index.tsx

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  EuiBadge,
+  EuiButtonEmpty,
+  EuiFlexGrid,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiPopover,
+  EuiPopoverFooter,
+  copyToClipboard,
+  useEuiTheme,
+} from '@elastic/eui';
+import { css } from '@emotion/react';
+import { i18n } from '@kbn/i18n';
+import React, { useState } from 'react';
+
+interface Props {
+  property: string;
+  value: string;
+  label?: string;
+  onFilter: () => void;
+}
+
+export function BadgeFilterWithPopover({ property, value, onFilter, label }: Props) {
+  const [isOpen, setIsOpen] = useState(false);
+  const theme = useEuiTheme();
+
+  return (
+    <EuiPopover
+      button={
+        <EuiBadge
+          data-test-subj="popoverButton"
+          color="hollow"
+          onClick={() => setIsOpen((state) => !state)}
+          onClickAriaLabel={i18n.translate(
+            'xpack.inventory.badgeFilterWithPopover.openPopoverBadgeLabel',
+            { defaultMessage: 'Open popover' }
+          )}
+        >
+          {label || value}
+        </EuiBadge>
+      }
+      isOpen={isOpen}
+      closePopover={() => setIsOpen(false)}
+    >
+      <span data-test-subj="popoverContentTitle">
+        <EuiFlexGroup
+          data-test-subj="popoverContent"
+          responsive={false}
+          gutterSize="xs"
+          css={css`
+            font-family: ${theme.euiTheme.font.familyCode};
+          `}
+        >
+          <EuiFlexItem grow={false}>
+            <span
+              css={css`
+                font-weight: bold;
+              `}
+            >
+              {property}:
+            </span>
+          </EuiFlexItem>
+          <EuiFlexItem grow={false}>
+            <span className="eui-textBreakWord">{value}</span>
+          </EuiFlexItem>
+        </EuiFlexGroup>
+      </span>
+      <EuiPopoverFooter>
+        <EuiFlexGrid responsive={false} columns={2}>
+          <EuiFlexItem>
+            <EuiButtonEmpty
+              data-test-subj="inventoryBadgeFilterWithPopoverFilterForButton"
+              iconType="plusInCircle"
+              onClick={onFilter}
+            >
+              {i18n.translate('xpack.inventory.badgeFilterWithPopover.filterForButtonEmptyLabel', {
+                defaultMessage: 'Filter for',
```

Comment:
Did you intentionally leave out the "filter out" option? thh, I prefer it this way because users can remove it from the filter at the top. It's more about consistency.

Reply:
Yes, I mentioned the reason [here](https://github.com/elastic/kibana/issues/192669#issuecomment-2371042668). In summary, we can filter for items and we can NOT filter OUT items.

Reply:
sorry I missed that 🙈